### DISCUSSION TITLE
TShock 5.1.1 for Terraria 1.4.4.8 (ζ Ori)

### DISCUSSION THREAD

**Discussion by hakusaro** [Pryaxis/TShock#2822] [Announcements]
Welcome to TShock for Terraria 5.1.1 with support for Terraria 1.4.4.8.

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

# Data loss warning

If you use server-side characters, we recommend enabling login before join. We discovered a bug that existed in the 5.0.0 release just before this release. If a player is disabled by TShock before they login, a situation can occur where all SSC data is erased. We suggest alerting your users to this fact and telling them not to login if they're disabled. If they do this, they may lose all character data in the database. Alternatively, enable login before join.

We hope to fix this in the future but we learned that this bug is present in 5.0.0 and therefore, not a blocker for this release for Terraria compatibility reasons.

## Notable changes in this release
* Fixed item giving potentially dropping too many items. (@PotatoCider, @punchready)
* Excluded GeoIP.dat from release bundle. (@SignatureBeef)
* Added `TownSlimeRed` to `FishableNpcIDs` list, allowing it to be fished up. (@drunderscore)
* Bump to Terraria 1.4.4.8 via OTAPI 3.1.18. (@hakusaro, @SignatureBeef)
  * In this version of Terraria, `Main.maxBuffTypes` and other `maxWhateverTypes` fields have been removed. Their replacements are in `Terraria.ID.whateverID.Count`. TShock calls to these fields have been swapped in order to bring forward compatibility with Terraria 1.4.4.8.
* In OTAPI 3.1.17, allowed Crystal Shard to grow. (@sgkoishi, @cc004, SignatureBeef/Open-Terraria-API#96)
* Added permission for summoning Mechdusa, Deerclops and slime pet. (@sgkoishi, #2808)
* Changed login to only restrict CC'd players during login whilst SSC is enabled. (@drunderscore)
  * This change allows the config option `RequireLogin` to function usefully again when SSC is not enabled.
* Changed `PlayerData.RestoreCharacter` to remove all buffs. (@drunderscore)
  * Before this change, it was theoretically possible to smuggle buffs onto servers with SSC enabled, by using buff indexes past `22`.
* Allowed Torch God's Favor to place different types of torches and campfires. (@sgkoishi, #2811)
* Updated translations! Currently, the major projects are at the following completion rates:
  * Chinese (93%)
  * Portuguese, Brazilian (89%)
  * Indonesian (89%)
  * Russian (56%)
  * Spanish (24%)
  * toki pona (10%)
  * Turkish (8%)
  * For complete credits, see the table below.

### Translation credits
The following translators contributed changes from November 1, 2022 to November 9, 2022 (from TShock 5.0.0 to 5.1.0).

|Contributor | Language | Translated words|
|-----|-----|-----|
|Shiva Goddess (ShivaGoddess) | Portuguese, Brazilian | 6979|
|Ricko (Rickojp) | Portuguese, Brazilian | 3154|
|RidnRaven (ridwankun2) | Indonesian | 2329|
|Janet Blackquill (pontaoski) | toki pona | 1216|
|SGKoishi | Chinese Simplified | 640|
|Cristofer GamerTVH (cristoferherame) | Spanish | 622|
|HDSeventh (hdseventh) | Indonesian | 315|
|EMRE ÇELİK (emre0447) | Turkish | 312|
|PHPoenX | Russian | 297|
|./lemon.sh (lemon-sh) | Polish | 206|
|Сергей Червяков (chsergeyg) | Russian | 182|
|okaythisisepic | Russian | 80|
|KomashiOFC | Portuguese, Brazilian | 76|
|Runesicle | toki pona | 22|
|Marotheit | Pirate English | 7|
|ATFGK | Chinese Simplified | 4|

## Upgrading

**Before installing, make a complete backup of your existing server and all files and folders, including `tshock.sqlite` and the `tshock` folder. We are not responsible for failed installations. Please make sure you backup your data. Severe data loss or gain may occur.**

To upgrade TShock, replace all of the files in the location of your server installation with all of the files from the release zip we provide. Overwrite all files. No data will be lost.

## Note about version 5.1.0

TShock version mastering occurs as the last step in the release process. When we cut a new version, we expect to release that version and all changes made prior to the version being cut. Unfortunately, a late-breaking bug with new internationalization changes between 5.0.0 and 5.1.0 was discovered after 5.1.0 was mastered. Therefore, we opted to tick the version _again_ after fixing the bug, and we mastered this release instead. No download for 5.1.0 will be made available, nor will a release.

<hr /><em>This discussion was created from the release <a href='https://github.com/Pryaxis/TShock/releases/tag/v5.1.1'>TShock 5.1.1 for Terraria 1.4.4.8 (ζ Ori)</a>.</em>

**Comment by TheCToTheY** [comment]
Hello. I'm a freelance translator, and I'd be glad to volunteer to translate TShock to Turkish in my free time, as thanks to keeping TShock alive and well. How can I get in touch to do this?

**Reply by sgkoishi** [reply]
You can just click the `Localized 23%` icon - or here's [the link to Turkish](https://crowdin.com/translate/tshock/6/en-tr)

**Comment by romanbanan123** [comment]
Hello, I have a problem, a TShock file.Server.exe it just doesn't open, the antivirus is turned off and doesn't start anyway

**Reply by Arthri** [reply]
You need the .NET 6 Console Runtime([Windows](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-6.0.11-windows-x64-installer), [macOS](https://dotnet.microsoft.com/en-us/download/dotnet/thank-you/runtime-6.0.11-macos-x64-installer))
